pkg/chunk/cache/memcached.go

```diff
@@ -211,9 +212,12 @@ func (c *Memcached) Store(ctx context.Context, keys []string, bufs [][]byte) {
 			return c.memcache.Set(&item)
 		})
 		if err != nil {
-			sp := opentracing.SpanFromContext(ctx)
-			sp.LogFields(otlog.Error(err))
 			level.Error(util.Logger).Log("msg", "failed to put to memcached", "err", err)
+
+			sp := opentracing.SpanFromContext(ctx)
+			if sp != nil {
+				sp.LogFields(otlog.Error(err))
```

Comment:
As far as I can see this is already done by `TimeRequestHistogramStatus()`
(not a change in this PR, but since you're making it longer)